packages/react/src/components/Callout/CalloutContent.base.tsx

```diff
@@ -204,69 +204,50 @@ function usePositions(
-                const currentPositionProps: IPositionProps = {
-                  ...props,
-                  target: targetRef.current!,
-                  bounds: getBounds(),
-                };
-
-                // If there is a finalHeight given then we assume that the user knows and will handle
-                // additional positioning adjustments so we should call positionCard
-                const newElementPositions: ICalloutPositionedInfo = finalHeight
-                  ? positionCard(currentPositionProps, hostElement.current, calloutElement.current, elementPositions)
-                  : positionCallout(
-                      currentPositionProps,
-                      hostElement.current,
-                      calloutElement.current,
-                      elementPositions,
-                    );
-
-                setCalloutPositions(newElementPositions);
+        // If we expect a target element to position against, `targetRef.current` should be defined.
+        // Once provided we can try to position the element.
+        const expectsTarget = !!target;
+
+        if (calloutElement.current) {
+          const cleanupObserver = observeResize(calloutElement.current, () => {
+            if (hostElement.current && calloutElement.current && (!expectsTarget || targetRef.current)) {
+              const currentPositionProps: IPositionProps = {
+                ...props,
+                target: targetRef.current!,
+                bounds: getBounds(),
+              };
+
+              // If there is a finalHeight given then we assume that the user knows and will handle
+              // additional positioning adjustments so we should call positionCard
+              const newElementPositions: ICalloutPositionedInfo = finalHeight
+                ? positionCard(currentPositionProps, hostElement.current, calloutElement.current, elementPositions)
+                : positionCallout(currentPositionProps, hostElement.current, calloutElement.current, elementPositions);
+
+              //Sets the current position of Callout upon resize. Callout will resize when:
+              //
+              // 1. `elementPositions` is undefined (initial render)
+              // 2. `elementPositions` does not equal the `newElementPositions`
+              // 3. There is a two decimal place difference in position.
+              if (
+                (elementPositions === undefined && newElementPositions) ||
+                (elementPositions && newElementPositions && !arePositionsEqual(elementPositions, newElementPositions))
+              ) {
+                onPositioned?.(newElementPositions);
+                setElementPositions(newElementPositions);
               }
-            });
-          }
-        }, calloutElement.current);
+            }
+          });
 
-        return () => async.cancelAnimationFrame(timerId);
+          return () => cleanupObserver();
```

Comment:
you should just be able to return `cleanupObserver` without making a lambda that just calls it.